vars/deployCloudFunction.groovy

```diff
@@ -0,0 +1,81 @@
+def call(config) {
+    def functionName = config.functionName
+    def projectId = config.projectId
+    def serviceAccountEmail = config.serviceAccountEmail
+    def zipFilePath = config.zipFilePath
+    def region = config.region
+    def environmentVariables = config.environmentVariables
+    def credentialsId = config.credentialsId
+    def runtime = config.runtime ?: 'python312'
+    def entryPoint = config.entryPoint
+    def timeout = config.timeout
+    def maxInstances = config.maxInstances
+    def allowUnauthenticated = config.allowUnauthenticated ?: false
+
+    // Validation of required parameters
+    if (!functionName || !projectId || !serviceAccountEmail || !zipFilePath || !region) {
+        error "Missing required parameters for deploying Cloud Function."
+    }
+
+    cdNode {
+        stage("Deploy Cloud Function: ${functionName}") {
+            container('gcloud') {
+                // TODO: tried adding ADC functionality but can't find docs to specify if this is possible
+                withCredentials([file(credentialsId: credentialsId, variable: 'GCLOUD_KEY')]) {
```

Comment:
It might be worth seeing if we could rather handle auth using Workload identity & our node service account: https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity

Reply:
Would I be able to test it if we use this service account or is that a permission blockade?

Reply:
Wait, I figured out how to use ADC; if we do not explicitly specify credentials, we use the ADC of the logged in user.

Reply:
We can als otry the [Google OAuth Credentials plugin](https://plugins.jenkins.io/google-oauth-plugin/)